app/services/forest_liana/oidc_client_manager.rb

```diff
@@ -4,22 +4,20 @@ module ForestLiana
   class OidcClientManager
     def self.get_client_for_callback_url(callback_url)
       begin
-        client_data = Rails.cache.read(callback_url) || nil
-        if client_data.nil?
-          configuration = ForestLiana::OidcConfigurationRetriever.retrieve()
-
-          if ForestLiana.forest_client_id.nil?
+        configuration = ForestLiana::OidcConfigurationRetriever.retrieve()
+        if ForestLiana.forest_client_id.nil?
+          client_data = Rails.cache.read("#{callback_url}-#{ForestLiana.env_secret}") || nil
```

Comment:
Maybe you should also suffix this key by the string '-clientId' just to indicate clearly what is the purpose of this value